server/src/db/lifecycle/compact_object_store.rs

```diff
@@ -372,7 +408,90 @@ async fn persist_stream_to_chunk<'a>(
         .context(ParquetChunkError)?,
     );
 
-    Ok(parquet_chunk)
+    Ok(Some(parquet_chunk))
+}
+
+/// Update the preserved catalog : replace compacted chunks with a newly persisted chunk
+async fn update_preserved_catalog(
+    db: &Db,
+    commpated_parquet_file_paths: &[ParquetFilePath],
+    parquet_chunk: &Option<Arc<ParquetChunk>>,
+) -> Result<()> {
+    // Open transaction
+    let mut transaction = db.preserved_catalog.open_transaction().await;
+
+    // Remove compacted chunks
+    for parquet_file_path in commpated_parquet_file_paths {
+        transaction.remove_parquet(parquet_file_path);
+    }
+
+    // Add new chunk if compaction returns some data
+    if let Some(parquet_chunk) = parquet_chunk {
+        let catalog_parquet_info = CatalogParquetInfo {
+            path: parquet_chunk.path().clone(),
+            file_size_bytes: parquet_chunk.file_size_bytes(),
+            metadata: parquet_chunk.parquet_metadata(),
+        };
+        transaction.add_parquet(&catalog_parquet_info);
+    }
+
+    // Close/commit the transaction
+    transaction.commit().await.context(CommitError)?;
+
+    Ok(())
+}
+
+async fn update_in_memory_catalog(
+    chunk_ids: &[ChunkId],
+    iox_metadata: IoxMetadata,
+    parquet_chunk: Option<Arc<ParquetChunk>>,
+    partition: Arc<RwLock<Partition>>,
+    delete_predicates_before: HashSet<Arc<DeletePredicate>>,
+) -> Result<Option<Arc<DbChunk>>> {
```

Comment:
```suggestion
) -> Option<Arc<DbChunk>> {
```

This should be infallible